src/sql/workbench/contrib/notebook/browser/notebook.component.ts

```diff
@@ -118,6 +120,20 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		this._register(this._configurationService.onDidChangeConfiguration(e => {
 			this.doubleClickEditEnabled = this._configurationService.getValue('notebook.enableDoubleClickEdit');
 		}));
+		this._register(DOM.addDisposableListener(window, DOM.EventType.KEY_DOWN, (e: KeyboardEvent) => {
+			// Check that there are no active elements to prevent the execution of undo/redo commands in other elements.
+			// We need to make sure that there are no active cells to prevent triggering undo/redo within code cells and check that text cells are not in edit mode.
+			if (this.isActive() && !this.areElementsActive() && (this.activeCellId === '' || !this.cellToolbar.first.getEditCellAction().editMode)) {
+				let event = new StandardKeyboardEvent(e);
+				if ((event.metaKey && event.shiftKey && event.keyCode === KeyCode.KEY_Z) || event.ctrlKey && event.keyCode === KeyCode.KEY_Y) {
+					event.stopPropagation();
```

Comment:
You should use `EventHelper.stop(event, true)` to stop both propagation and cancel default